course5/homework9/unique_ptr/main.cc

```diff
@@ -0,0 +1,54 @@
+
+#include "unique_ptr.h"
+
+#include <iostream>
+#include <cassert>
+
+class TestClass
+{
+public:
+	explicit TestClass(int instanceNo): mInstanceNo(instanceNo)
+	{
+		std::cout << "Creating instance " << mInstanceNo << std::endl;
+		++mInstanceCount;
```

Comment:
To get this to work, you should also increase the counter in the copy constructor. This will you could detect any accidental copy.